packages/block-editor/src/hooks/duotone.js

```diff
@@ -258,50 +307,28 @@ const withDuotoneStyles = createHigherOrderComponent(
 			props.name,
 			'color.__experimentalDuotone'
 		);
-		const duotonePalette = useMultiOriginPresets( {
-			presetSetting: 'color.duotone',
-			defaultSetting: 'color.defaultDuotone',
-		} );
-
-		const id = `wp-duotone-${ useInstanceId( BlockListBlock ) }`;
-
-		let colors = props?.attributes?.style?.color?.duotone;
-
-		if ( ! Array.isArray( colors ) ) {
-			const duotone = duotonePalette.find( ( dt ) => dt.slug === colors );
 
-			if ( duotone ) {
-				colors = duotone.colors;
-			}
-		}
+		const id = `wp-duotone-flibble-${ useInstanceId( BlockListBlock ) }`;
+		const className = classnames( props?.className, id );
+		const duotoneStyle = props?.attributes?.style?.color?.duotone;
 
-		if ( ! duotoneSupport || ! colors ) {
+		// CAUTION: code added before the early return will be executed
+		// for all blocks, not just those that support duotone. Code added
+		// above this line should be carefully evaluated for its impact on
+		// performance.
+		if ( ! duotoneSupport ) {
 			return <BlockListBlock { ...props } />;
 		}
 
-		// Extra .editor-styles-wrapper specificity is needed in the editor
-		// since we're not using inline styles to apply the filter. We need to
-		// override duotone applied by global styles and theme.json.
-		const selectorsGroup = scopeSelector(
-			`.editor-styles-wrapper .${ id }`,
-			duotoneSupport
-		);
-
-		const className = classnames( props?.className, id );
-
-		const element = useContext( BlockList.__unstableElementContext );
-
 		return (
 			<>
-				{ element &&
-					createPortal(
-						<InlineDuotone
-							selector={ selectorsGroup }
-							id={ id }
-							colors={ colors }
-						/>,
-						element
-					) }
+				{ duotoneStyle && (
```

Comment:
Can we also just remove the early return now and add the `duotoneSupport` check here too.
I guess we need a check for the `classname` prop too.

Reply:
Ok sure. 

So I can understand (humour me)...

We are doing this for performance reasons around allowing React to handle reconciliation (which I've read up on and understand more about now).

But (IMHO) this does make the code harder to reason about and easier to introduce errors. An early `return` sets out a single clear point beyond which no inconsistencies should be introduced.

With the approach of a JSX conditional, we have to deal with things like conditionalising the `className` (one example) which could lead to a whole class of bugs - especially as we have no tests.

What I'm curious to understand is if the benefits of allowing React to handle reconciliation of the _duotone support_ element (which I believe won't change once the block has rendered) outweighs the potential (and yes, I  know it's subjective) benefits of clearer code.

I'm asking this not as a critique of your suggestion, but more to understand how you personally evaluate such decisions.

Reply:
PS: I made the change